js/main.js

```diff
@@ -1,2 +1,70 @@
 /* global data */
 /* exported data */
+
+var $entryForm = document.querySelector('[data-view="entry-form"]');
+var $entryUrl = document.querySelector('#entry-url');
+var $entryTitle = document.querySelector('#entry-title');
+var $entryNotes = document.querySelector('#entry-notes');
+var $entryImage = document.querySelector('.entry-image > img');
+var photoUrl = '';
+var photoTitle = '';
+var notes = '';
+
+if (localStorage.getItem('data')) {
+  for (var key in data) {
+    var previousDataJSON = localStorage.getItem('data');
+    var previousData = JSON.parse(previousDataJSON);
+    data[key] = previousData[key];
+  }
+}
+
+$entryForm.addEventListener('input', function (event) {
+  if (event.target === $entryUrl) {
+    photoUrl = event.target.value;
+  }
+  if (event.target === $entryTitle) {
+    photoTitle = event.target.value;
+  }
+  if (event.target === $entryNotes) {
+    notes = event.target.value;
+  }
+});
```

Comment:
I don't see a need to make these global variables. You only need to worry about these values when an entry is submitted.

Reply:
I used them to store the values for later. You're right that I mostly used them in the listening handle function for the submit event, but I also update the picture src, title and alt as well (in my blur event listener). Should I not have bothered with that?